src/kili/services/export/format/coco/common.py

```diff
@@ -0,0 +1,273 @@
+"""
+Common code for the coco exporter.
+"""
+
+import json
+import os
+import shutil
+import time
+from datetime import datetime
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Dict, List, Tuple
+
+import numpy as np
+from PIL import Image
+from tqdm.autonotebook import tqdm
+from typing_extensions import TypedDict
+
+from kili.orm import Asset
+from kili.services.export.format.base import BaseExporter
+from kili.services.export.types import Job, JobName, Jobs, ProjectId
+
+DATA_SUBDIR = "data"
+
+
+# COCO format
+class _CocoImage(TypedDict):
+    id: int
+    license: int
+    file_name: str
+    height: int
+    width: int
+    date_captured: None
+
+
+class _CocoCategory(TypedDict):
+    id: int
+    name: str
+    supercategory: str
+
+
+class _CocoAnnotation(TypedDict):
+    id: int
+    image_id: int
+    category_id: int
+    bbox: List[int]
+    segmentation: List[List[float]]  # [[x, y, x, y, x ...]]
+    area: int
+    iscrowd: int
+
+
+class _CocoFormat(TypedDict):
+    info: Dict  # type: ignore
+    licenses: List[Dict]  # type: ignore
+    categories: List[_CocoCategory]
+    images: List[_CocoImage]
+    annotations: List[_CocoAnnotation]
+
+
+class CocoExporter(BaseExporter):
+    """
+    Common code for Kili exporters.
+    """
+
+    download_media = True
```

Comment:
why do you need to download the asset here ?

Reply:
COCO is self-contained and requires paths to the media files
(comment added)